tests/framework/resourcemanager.go

```diff
@@ -0,0 +1,252 @@
+}
+
+func (rm *ResourceManager) readAndHandleObjects(
+	handle func(unstructured.Unstructured) error,
+	files []string,
+) error {
+	for _, file := range files {
+		data, err := rm.getFileContents(file)
+		if err != nil {
+			return err
+		}
+
+		decoder := yaml.NewYAMLOrJSONDecoder(data, 4096)
+		for {
+			obj := unstructured.Unstructured{}
+			if err := decoder.Decode(&obj); err != nil {
+				if errors.Is(err, io.EOF) {
+					break
+				}
+				return fmt.Errorf("error decoding resource: %w", err)
+			}
+
+			if len(obj.Object) == 0 {
+				continue
+			}
+
+			if err := handle(obj); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// getContents takes a string that can either be a local file
+// path or an https:// URL to YAML manifests and provides the contents.
+func (rm *ResourceManager) getFileContents(file string) (*bytes.Buffer, error) {
+	if strings.HasPrefix(file, "http://") {
+		return nil, fmt.Errorf("data can't be retrieved from %s: http is not supported, use https", file)
+	} else if strings.HasPrefix(file, "https://") {
+		ctx, cancel := context.WithTimeout(context.Background(), rm.TimeoutConfig.ManifestFetchTimeout)
+		defer cancel()
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, file, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+
+		manifests := new(bytes.Buffer)
+		count, err := manifests.ReadFrom(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.ContentLength != -1 && count != resp.ContentLength {
+			return nil, fmt.Errorf("received %d bytes from %s, expected %d", count, file, resp.ContentLength)
+		}
+		return manifests, nil
+	}
+
+	if !strings.HasPrefix(file, "manifests/") {
+		file = "manifests/" + file
+	}
+
+	b, err := rm.FS.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	return bytes.NewBuffer(b), nil
+}
+
+// WaitForAppsReady waits for all apps in the specified namespace to be ready, or until the ctx timeout is reached.
+func (rm *ResourceManager) WaitForAppsReady(k8sClient client.Client, namespace string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rm.TimeoutConfig.CreateTimeout)
+	defer cancel()
+
+	return wait.PollUntilContextCancel(
+		ctx,
+		500*time.Millisecond,
+		true, /* poll immediately */
+		func(ctx context.Context) (bool, error) {
+			// first check for Pods to be ready
+			var podList core.PodList
+			if err := k8sClient.List(ctx, &podList, client.InNamespace(namespace)); err != nil {
+				return false, err
+			}
+
+			var podsReady int
+			for _, pod := range podList.Items {
+				for _, cond := range pod.Status.Conditions {
+					if cond.Type == core.PodReady && cond.Status == core.ConditionTrue {
+						podsReady++
+					}
+				}
+			}
+
+			if podsReady != len(podList.Items) {
+				return false, nil
+			}
+
+			// now check for Gateway to be programmed
```

Comment:
I wanted to ensure that we're checking a status that tells us that nginx is ready. HTTPRoutes don't have such a status.

Reply:
but at least we can check that they have been accepted, no?

Reply:
I can add that if we think it has increased value over the readiness of the Gateway

Reply:
This might catch a bug if HTTPRoute status is not populated in some cases. 

This way our tests can check:
(1) Ensure control plane transactions (changes to resources) are handled correctly -- the statuses of the corresponding resources are updated
(2) Ensure traffic flows through data plane configured according to the resources.

So receiving 200 conforms  (2), but doesn't fully confirm (1), because the lack of status check.

Reply:
Are we redoing what the conformance tests are doing though?

Reply:
but the same thing can be said about checking Gateway statuses?

Reply:
my goal with these checks was purely ensure we are ready before we send traffic. It wasn't to verify that we're handling resources properly. Now, if those checks fail, it could inform us of a bug, but that's not what the initial intention is for.